pyproject.toml

```diff
@@ -134,6 +105,33 @@ exclude = [
     "doc/_build",
 ]
 
+[tool.uv]
+dev-dependencies = [
```

Comment:
this should wait on https://github.com/astral-sh/uv/issues/5632, i.e. we shouldn't need to collapse all the dev dependencies into a single list

Reply:
could do. This might not be premature optimisation though as the current dev dependencies-

- install *very* quickly using UV
- have no conflicts

based on that i think splitting them into separate groups might be a micro-optimisation

Reply:
Nothing to do with speed: you are forcing all test environments to have unnecessary resolve constraints against unrelated packages, which can change the versions of packages you get

Reply:
yep, this is true in the general case. *that* also introduces the risk that different contributors resolve different versions depending on whether they are also building the docs, for example. Since even with dependency groups, the groups are not isolated from each other

Reply:
As of [0.4.27](https://github.com/astral-sh/uv/releases/tag/0.4.27), uv supports separate groups. They still will constrain each other though, as the lockfile is for the entire project.